core/src/test/scala/org/apache/spark/partial/SumEvaluatorSuite.scala

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.partial
+
+
+import org.apache.spark._
+import org.apache.spark.util.StatCounter
+
+class SumEvaluatorSuite extends SparkFunSuite with SharedSparkContext {
+
+  test("correct handling of count 1") {
+
+    //setup
+    val counter = new StatCounter(List(2.0))
+    // count of 10 because it's larger than 1,
+    // and 0.95 because that's the default
+    val evaluator = new SumEvaluator(10, 0.95)
+    // arbitrarily assign id 1
+    evaluator.merge(1, counter)
+
+    //execute
+    val res = evaluator.currentResult()
+    // 38.0 - 7.1E-15 because that's how the maths shakes out
+    val target_mean =  38.0 - 7.1E-15
+    // Version to test everything except mean
+    val ancillary_res = new BoundedDouble(target_mean, res.confidence, res.low, res.high)
+    // mean result to test
+    val mean_diff = (res.mean - target_mean).abs
+    
+
+    //Sanity check that equality works on BoundedDouble
+    assert(new BoundedDouble(2.0, 0.95, 1.1, 1.2) == new BoundedDouble(2.0, 0.95, 1.1, 1.2))
+    // actual test
+
+    // Use ancillary_res as the result to check everything except the mean
+    assert(ancillary_res == new BoundedDouble(target_mean, 0.950, Double.NegativeInfinity, Double.PositiveInfinity))
+
+    // check that mean is within expected tolerance of expectation
+    assert(mean_diff < Double.MinPositiveValue)
```

Comment:
This is almost like just asserting equality. If we're this sure about the result then I assume we're actually sure of exact equality, and so would be fine to assert that directly for clarity.

To assert approximate equality you can see the other tests that use syntax like `assert(a ~== b absTol 1e-7)`. Approximate equality assertions are likely to involve asserting some more possible variance than the smallest possible double. Where it's necessary I'd expect you need wider tolerance; where it's not necessary, it should just be an exact equality assertion.